bot/cogs/news.py

```diff
@@ -0,0 +1,228 @@
+                if message is None:
+                    log.warning("Can't fetch PEP new message ID.")
+                    continue
+            pep_news.append(message.embeds[0].title)
+
+        # Reverse entries to send oldest first
+        data["entries"].reverse()
+        for new in data["entries"]:
+            try:
+                new_datetime = datetime.strptime(new["published"], "%a, %d %b %Y %X %Z")
+            except ValueError:
+                log.warning(f"Wrong datetime format passed in PEP new: {new['published']}")
+                continue
+            if (
+                    new["title"] in pep_news
+                    or new_datetime.date() < date.today()
+            ):
+                continue
+
+            msg_id = await self.send_webhook(
+                title=new["title"],
+                description=new["summary"],
+                timestamp=new_datetime,
+                url=new["link"],
+                webhook_profile_name=data["feed"]["title"]
+            )
+            payload["data"]["pep"].append(msg_id)
+
+        # Apply new sent news to DB to avoid duplicate sending
+        await self.bot.api_client.put("bot/bot-settings/news", json=payload)
+
+    @loop(minutes=20)
+    async def post_maillist_news(self) -> None:
+        """Send new maillist threads to #python-news that is listed in configuration."""
+        await self.bot.wait_until_guild_available()
+        await self.sync_maillists()
+        existing_news = await self.bot.api_client.get("bot/bot-settings/news")
+        payload = existing_news.copy()
+
+        for maillist in constants.PythonNews.mail_lists:
+            async with self.bot.http_session.get(RECENT_THREADS_TEMPLATE.format(name=maillist)) as resp:
+                recents = BeautifulSoup(await resp.text())
+
+            for thread in recents.html.body.div.find_all("a", href=True):
+                # We want only these threads that have identifiers
+                if "latest" in thread["href"]:
+                    continue
+
+                thread_information, email_information = await self.get_thread_and_first_mail(
+                    maillist, thread["href"].split("/")[-2]
+                )
+
+                try:
+                    new_date = datetime.strptime(email_information["date"], "%Y-%m-%dT%X%z")
+                except ValueError:
+                    log.warning(f"Invalid datetime from Thread email: {email_information['date']}")
+                    continue
+
+                if (
+                        await self.check_new_exist(thread_information["subject"], new_date, maillist, existing_news)
+                        or new_date.date() < date.today()
+                ):
+                    continue
+
+                content = email_information["content"]
+                link = THREAD_URL.format(id=thread["href"].split("/")[-2], list=maillist)
+                msg_id = await self.send_webhook(
```

Comment:
We should publish these messages.

Once sent they should be fetched using this ID and the `.publish()` method should be used to crosspost this to any subscribing servers.

Reply:
Resolved in 547de1a , but I made `send_webhook` returning message and publish this message.